CMakeLists.txt

```diff
@@ -192,7 +194,11 @@ else()
         set(SAM_NAME SAM)
     endif()
     if(APPLE)
-        set(SAM_APP ${CMAKE_CURRENT_BINARY_DIR}/${SAM_NAME}.app)
+  	    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+	        set(SAM_APP ${CMAKE_CURRENT_BINARY_DIR}/Debug/${SAM_NAME}.app)
+	     else()
+	        set(SAM_APP ${CMAKE_CURRENT_BINARY_DIR}/Release/${SAM_NAME}.app)
+	     endif()
```

Comment:
There should be two seprate folders build/Debug and build/Release depending on what configuration is set in the top level CMakeLists.txt that was included in the zip file. I could not get the "Debug,Release" combined setting to work with XCode how we have working for VS Studio. The issue is in cmake_xcode_clean.sh, the arguments
-DCMAKE_BUILD_TYPE="Debug" -DSAM_SKIP_TESTS=1 "-DCMAKE_OSX_ARCHITECTURES=x86_64;arm64" -DCMAKE_XCODE_ATTRIBUTE_ONLY_ACTIVE_ARCH=NO
only work for one build type (Debug or Release) but not both with the Architecture arguments in CMake. The CMAKE_CONFIGURATION_TYPES "Debug;Release" fails to work with CMake on macOS.

So, the build/Release folder has a valid SAM.app and the build/Debug has a valid SAMd.app.
Both debug and release folders appear under the build/Release/sam/ folder but only the release folder is valid and the same with the debug folder.

Reply:
Right, so I have cmake-build-release instead of build/Release and cmake-build-debug instead of build/Debug and each of them generate their own cmake files depending on CMAKE_BUILD_TYPE. Each of those folders has their own ${CMAKE_CURRENT_BINARY_DIR} which will be build/Release/sam or cmake-build-release/sam.

It's the nested Release folder within CMAKE_CURRENT_BINARY_DIR that is breaking the workflow for me. If I just run CMake via command line (which I need to do for PySAM builds on both Mac and Linux), by doing this (not I am not generating GXcode):

```
mkdir cmake-build-debug
cd cmake-build-debug
cmake .. -DSAMAPI_EXPORT=0 -DCMAKE_BUILD_TYPE="Debug" -DSAM_SKIP_TESTS=1 " -DCMAKE_OSX_ARCHITECTURES=x86_64;arm64"
make -j8
```

This fails:
```
[100%] Linking CXX executable SAMd.app/Contents/MacOS/SAMd
building file list ... done
rsync: mkdir "/Users/dguittet/SAM-Dev/build-debug/sam/Debug/SAMd.app/Contents" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at /AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/rsync/rsync-54.120.1/rsync/main.c(545) [receiver=2.6.9]
rsync: connection unexpectedly closed (8 bytes received so far) [sender]
rsync error: error in rsync protocol data stream (code 12) at /AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/rsync/rsync-54.120.1/rsync/io.c(453) [sender=2.6.9]
make[2]: *** [sam/SAMd.app/Contents/MacOS/SAMd] Error 12
make[2]: *** Deleting file `sam/SAMd.app/Contents/MacOS/SAMd'
make[1]: *** [sam/CMakeFiles/SAMd.dir/all] Error 2
make: *** [all] Error 2
```

It seems like Xcode is enforcing this nested Release folder, whereas the CMake cmd line does not. Can you try this:

```
    if(APPLE)
        set(SAM_APP ${CMAKE_CURRENT_BINARY_DIR}/${SAM_NAME}.app)
        set_target_properties(${SAM_EXE}
                PROPERTIES
                RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
    else()
```
at line 197 and see if that works in Xcode? It should push the SAM.app into build/Release/sam and SAMd.app into build/Debug/sam

Reply:
With this update, the cmake XCode generation fails with:
[100%] Linking CXX executable SAMd.app/Contents/MacOS/SAMd
building file list ... done
rsync: mkdir "/Users/dguittet/SAM-Dev/build-debug/sam/Debug/SAMd.app/Contents" failed: No such file or directory (2)
rsync error: error in file IO (code 11) at /AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/rsync/rsync-54.120.1/rsync/main.c(545) [receiver=2.6.9]
rsync: connection unexpectedly closed (8 bytes received so far) [sender]
rsync error: error in rsync protocol data stream (code 12) at /AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/rsync/rsync-54.120.1/rsync/io.c(453) [sender=2.6.9]
make[2]: *** [sam/SAMd.app/Contents/MacOS/SAMd] Error 12
make[2]: *** Deleting file `sam/SAMd.app/Contents/MacOS/SAMd'
make[1]: *** [sam/CMakeFiles/SAMd.dir/all] Error 2
make: *** [all] Error 2

Reply:
Updated as follows CMakeLists.txt line 196:
    if(APPLE)
    	if (CMAKE_GENERATOR STREQUAL Xcode)
	  	    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
		        set(SAM_APP ${CMAKE_CURRENT_BINARY_DIR}/Debug/${SAM_NAME}.app)
	    	else()
	        	set(SAM_APP ${CMAKE_CURRENT_BINARY_DIR}/Release/${SAM_NAME}.app)
	     	endif()
    	else()
        	set(SAM_APP ${CMAKE_CURRENT_BINARY_DIR}/${SAM_NAME}.app)
        	set_target_properties(${SAM_EXE}
                PROPERTIES
                RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
        endif()
Working XCode and command line. Please verify...